### PR TITLE
Remove mkdirp and replace with internal util

### PR DESCRIPTION
**Pull request by palmerj3**
Addresses the security vulnerability and keeps support for node 8.x

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 const xml = require('xml');
-const mkdirp = require('mkdirp');
+const makeDirRecursive = require('./utils/makeDir').makeDirRecursive;
```
Comment (phenomnomnominal) [inline index.js]: `const { makeDirRecursive } = require('./utils/makeDir')` ?

File: utils/makeDir.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+const fs = require('fs');
+const path = require('path');
+
+const makeDirRecursive = (pathToCreate) => {
+  const dirs = pathToCreate.split(path.sep);
+
+  let workingPath = '';
+
+  while (dirs.length > 0 && fs.existsSync(pathToCreate) === false) {
```
Comment (phenomnomnominal) [inline utils/makeDir.js]: Could this be `while (dirs.length > 0 && !fs.existsSync(pathToCreate)) {` ?

File: utils/makeDir.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    workingPath = path.join(workingPath, dirs.shift());
+
+    if (fs.existsSync(workingPath) === false) {
```
Comment (phenomnomnominal) [inline utils/makeDir.js]: `if (!fs.existsSync(pathToCreate)) {` ?

File: utils/makeDir.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+const fs = require('fs');
+const path = require('path');
+
+const makeDirRecursive = (pathToCreate) => {
+  const dirs = pathToCreate.split(path.sep);
```
Comment (phenomnomnominal) [inline utils/makeDir.js]: Do you need to validate the path at all?

Comment (palmerj3) [reply utils/makeDir.js]: Validate how?

**Pull request closed without merging**